pkg/operator/controller/controller_lb.go

```diff
@@ -127,7 +127,7 @@ func (r *reconciler) finalizeLoadBalancerService(ci *operatorv1.IngressControlle
 	// an annotation on the ingresscontroller.
 	ingress := service.Status.LoadBalancer.Ingress
 	if len(ingress) > 0 && len(ingress[0].Hostname) > 0 {
-		records, err := desiredDNSRecords(ci, ingress[0].Hostname, dnsConfig)
+		records, err := desiredDNSAliasRecords(ci, ingress[0].Hostname, dnsConfig)
```

Comment:
Looks like this needs a similar treatment for `.IP` handling to avoid leaking A records